src/tracing/mux.rs

```diff
@@ -0,0 +1,408 @@
+            MultiInspector::CallInspector(_, inspector) => inspector.call(context, inputs),
+            MultiInspector::PrestateInspector(_, inspector) => inspector.call(context, inputs),
+            MultiInspector::NoopInspector => None,
+            MultiInspector::MuxInspector(inspector) => inspector.call(context, inputs),
+        };
+
+        None
+    }
+
+    #[inline]
+    fn call_end<DB: Database>(
+        &mut self,
+        context: &mut EvmContext<DB>,
+        inputs: &CallInputs,
+        outcome: CallOutcome,
+    ) -> CallOutcome {
+        match self {
+            MultiInspector::FourByteInspector(inspector) => {
+                inspector.call_end(context, inputs, outcome)
+            }
+            MultiInspector::CallInspector(_, inspector) => {
+                inspector.call_end(context, inputs, outcome)
+            }
+            MultiInspector::PrestateInspector(_, inspector) => {
+                inspector.call_end(context, inputs, outcome)
+            }
+            MultiInspector::NoopInspector => outcome,
+            MultiInspector::MuxInspector(inspector) => inspector.call_end(context, inputs, outcome),
+        }
+    }
+
+    #[inline]
+    fn create<DB: Database>(
+        &mut self,
+        context: &mut EvmContext<DB>,
+        inputs: &mut CreateInputs,
+    ) -> Option<CreateOutcome> {
+        match self {
+            MultiInspector::FourByteInspector(inspector) => inspector.create(context, inputs),
+            MultiInspector::CallInspector(_, inspector) => inspector.create(context, inputs),
+            MultiInspector::PrestateInspector(_, inspector) => inspector.create(context, inputs),
+            MultiInspector::NoopInspector => None,
+            MultiInspector::MuxInspector(inspector) => inspector.create(context, inputs),
+        };
+
+        None
+    }
+
+    #[inline]
+    fn create_end<DB: Database>(
+        &mut self,
+        context: &mut EvmContext<DB>,
+        inputs: &CreateInputs,
+        outcome: CreateOutcome,
+    ) -> CreateOutcome {
+        match self {
+            MultiInspector::FourByteInspector(inspector) => {
+                inspector.create_end(context, inputs, outcome)
+            }
+            MultiInspector::CallInspector(_, inspector) => {
+                inspector.create_end(context, inputs, outcome)
+            }
+            MultiInspector::PrestateInspector(_, inspector) => {
+                inspector.create_end(context, inputs, outcome)
+            }
+            MultiInspector::NoopInspector => outcome,
+            MultiInspector::MuxInspector(inspector) => {
+                inspector.create_end(context, inputs, outcome)
+            }
+        }
+    }
+
+    #[inline]
+    fn selfdestruct<DB: Database>(&mut self, contract: Address, target: Address, value: U256) {
+        match self {
+            MultiInspector::FourByteInspector(inspector) => {
+                <FourByteInspector as Inspector<DB>>::selfdestruct(
```

Comment:
I faced the same issue with the selfdestruct call but couldn't figure out why this is necessary here

Reply:
I think it's because the `FourByteInspector` implements the `Inspector<DB>` trait, where `selfdestruct` function has no reference to the `DB` type, and compiler can't infer the `Inspector` trait type